ee/watchdog/controller_windows.go

```diff
@@ -52,21 +81,76 @@ func NewController(ctx context.Context, k types.Knapsack, configFilePath string)
 		return nil, fmt.Errorf("opening log db in %s: %w", k.RootDirectory(), err)
 	}
 
-	return &WatchdogController{
-		slogger:        k.Slogger().With("component", "watchdog_controller"),
+	slogger := k.Slogger().With("component", "watchdog_controller")
+
+	tm := watchdogTaskManager{
+		slogger:        slogger,
 		knapsack:       k,
-		interrupt:      make(chan struct{}, 1),
-		logPublisher:   logPublisher,
 		configFilePath: configFilePath,
-	}, nil
+	}
+
+	tm.setCurrentInstallState(installStateUnset)
+
+	wc := &WatchdogController{
+		slogger:      slogger,
+		knapsack:     k,
+		interrupt:    make(chan struct{}, 1),
+		logPublisher: logPublisher,
+		taskManager:  &tm,
+	}
+
+	return wc, nil
 }
 
+// FlagsChanged is called when the LauncherWatchdogDisabled flag is seen from our control server.
+// It checks if the watchdog disabled flag has changed and enables or disables the watchdog task accordingly.
+// This is currently the only production path into watchdog installation or removal, so agents will not
+// install watchdog until they've received this flag.
 func (wc *WatchdogController) FlagsChanged(ctx context.Context, flagKeys ...keys.FlagKey) {
 	_, span := observability.StartSpan(ctx)
 	defer span.End()
 
-	if slices.Contains(flagKeys, keys.LauncherWatchdogEnabled) {
-		wc.ServiceEnabledChanged(wc.knapsack.LauncherWatchdogEnabled())
+	if !slices.Contains(flagKeys, keys.LauncherWatchdogDisabled) {
+		return
+	}
+
+	if !wc.shouldManageWatchdog() {
+		wc.slogger.Log(ctx, slog.LevelDebug,
+			"skipping watchdog manipulation for either non-prod setup or lack of admin privileges",
+		)
+		return
+	}
+
+	currentInstallState := wc.taskManager.currentInstallState()
+	nowDisabled := wc.knapsack.LauncherWatchdogDisabled()
+
+	if currentInstallState == installStateUnset {
+		taskExists, err := watchdogTaskExists(wc.knapsack.Identifier())
+		if !taskExists || err != nil { // if we err for any reason assume not installed
```

Comment:
Do we want to log the error here?

Reply:
oh that's not a bad idea, was thinking it would usually error for not installed but forgot we squash that in `watchdogTaskExists`. i'll add in a new PR, thanks!